streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsPartitionAssignor.java

```diff
@@ -86,10 +87,23 @@
 import static org.apache.kafka.streams.processor.internals.assignment.StreamsAssignmentProtocolVersions.UNKNOWN;
 import static org.apache.kafka.streams.processor.internals.assignment.SubscriptionInfo.UNKNOWN_OFFSET_SUM;
 
-public class StreamsPartitionAssignor implements ConsumerPartitionAssignor, Configurable {
+public class StreamsPartitionAssignor implements ConsumerPartitionAssignor, Configurable,
+    ContextualLogging {
```

Comment:
Isn't it a little weird that the assignor implement that contextual logging interface? It's an assignor, not a logger?

Reply:
Yeah, I was struggling to name the mixin nicely (the `Configurable` naming pattern would yield something silly like `ContexuallyLoggable`). After mulling on it, I think I'll change it to `DynamicLoggingContextMixin`. It's an internal interface for now, so I think simple and clear is more important than pretty for now.